mdp/nodes/sfa_nodes.py

```diff
@@ -159,7 +197,8 @@ def _stop_training(self, debug=False):
                            " or PCANode(svd=True)"% str(d))
                 raise NodeException(err_msg)
         except SymeigException as exception:
-            errstr = str(exception)+"\n Covariance matrices may be singular."
+            errstr = (str(exception)+"\n Covariance matrices may be singular."
+                    +"\n Try to create SFA node with handle_rank_deficit=True.")
```

Comment:
This error message should be a lot more verbose. Laurenz thinks that this error should say as much about what happens here as necessary for users to understand the problem. 

Here's the version from my PR, which might also not be ideal yet:
```
 except SymeigException as exception:
    errstr  = str(exception) + "\n Covariance matrices may be singular.\n"
    errstr += "This usually happens if there are redundancies in the (expanded) training data.\n" + \
              "There are several ways to deal with this issue:\n\n" + \
              "   - Use more data.\n\n" + \
              "   - Add noise to the data. This can be done by slotting an additional NoiseNode\n" + \
              "     in front of a troublesome SFANode. Noise levels do not have to be high.\n\n" + \
              "   - Run training data through PCA. This can be done by slotting an additional\n" + \
              "     PCA node in front of the troublesome SFANode. Use the PCA node to discard\n" + \
              "     dimensions within your data with lower variance.\n\n" + \
              "   - [ RECOMMENDED ] Use SVD by setting the SFANode flag use_svd to True. This\n" + \
              "     solves the internal SFA eigenvalue problem by using two instances of singular\n" + \
              "     value decomposition (SVD) instead of the default behavior of using the\n" + \
              "     systems's symeig() routine to solve a single, generalized eigenvalue problem.\n" + \
              "     Using SVD is slightly more expensive but more robust.\n"
    raise NodeException(errstr)
```
> Using SVD is slightly more expensive but more robust

At least this will probably have to be re-written once we have the full benchmarks of all the methods.

Reply:
I'm fine with having a more verbose error message, as long as it contains an easy instruction how to trivially  fix the issue.